src/plcc/load_spec/validate_spec/validate_syntactic_spec/build_parsing_table_test.py

```diff
@@ -0,0 +1,90 @@
+from pytest import raises
+from .build_parsing_table import build_parsing_table, Table
+from collections import defaultdict
+
+def test_empty():
+    table = build_parsing_table({}, {}, {})
+    assert len(table.getFilledCellLocations()) == 0
+
+def test_set_cell():
+    table = build_parsing_table({}, {}, {})
+    table.setCell('nonterminal', 'TERMINAL', ["PRODUCTION"])
+    assert table.getCell('nonterminal', 'TERMINAL') == ["PRODUCTION"]
+
+def test_left_recursive():
+    firstSets = {
+        "s" : {"","A","C","D"},
+        "x" : {"","A","C","D"},
+        "y" : {"","A"},
+        "z" : {"C","D"},
+        "" : {""},
+        "x y z" : {"A","C","D"},
+        "x s" : {"A", "C", "D"},
+        "A y B" : {"A"},
+        "C z" : {"C"},
+        "D" : {"D"}
+    }
+    followSets = {
+        "s" : {chr(26), "A","C","D"},
+        "x" : {"A","C","D"},
+        "y" : {"B","C","D"},
+        "z" : {chr(26), "A","C","D"}
+    }
+    rules = {
+        "s" : ["","x y z"],
+        "x" : ["","x s"],
+        "y" : ["","A y B"],
+        "z" : ["C z", "D"]
+    }
+    table = build_parsing_table(firstSets, followSets, rules)
+
+    assert table.getCell('s','A') == ["", "x y z"]
+    assert table.getCell('s','C') == ["", "x y z"]
+    assert table.getCell('s', 'D') == ["", "x y z"]
+    assert table.getCell('s', '\x1a') == [""]
+
+    assert table.getCell('x', 'A') == ["", "x s"]
+    assert table.getCell('x', 'C') == ["", "x s"]
+    assert table.getCell('x', 'D') == ["", "x s"]
+
+    assert table.getCell('y', 'A') == ["A y B"]
+    assert table.getCell('y', 'B') == [""]
+    assert table.getCell('y', 'C') == [""]
+    assert table.getCell('y', 'D') == [""]
+
+    assert table.getCell('z', 'C') == ["C z"]
+    assert table.getCell('z', 'D') == ["D"]
+    assert len(table.getFilledCellLocations()) == 13
+
+def test_complex():
```

Comment:
What do you mean by `complex`? What is this test testing that other tests are not? And what should be the result? Encode that in the test name and the implementation.